app/src/wavevr/cpp/DeviceDelegateWaveVR.cpp

```diff
@@ -350,6 +352,26 @@ struct DeviceDelegateWaveVR::State {
         delegate->EndTouch(controller.index);
       }
       delegate->SetAxes(controller.index, immersiveAxes, kNumAxes);
+      delegate->SetHapticsCount(controller.index, 1);
+
+      uint64_t  inputFrameID = 0;
+      float pulseDuration = 0.0f, pulseIntensity = 0.0f;
+      delegate->GetHapticsFeedback(controller.index, inputFrameID, pulseDuration, pulseIntensity);
+      if (inputFrameID == 0 || pulseDuration <= 0 ) {
+        // Stopping haptics feedback.
+        if (controller.doingVibration) {
+          WVR_TriggerVibration(controller.type, WVR_InputId_Max, 0.0f, 0);
+          controller.doingVibration = false;
+        }
+      } else if (pulseIntensity > 0.0f) {
+        controller.doingVibration = true;
+        // THe duration time unit needs to be transformed from milliseconds to microseconds.
+        // The gamepad extensions API does not yet have independent control
+        // of frequency and intensity. It only has vibration value (0.0 ~ 1.0).
+        // In this WaveVR SDK, the value makes more sense to be intensity because frequency can't
+        // < 1.0 here.
+        WVR_TriggerVibration(controller.type, WVR_InputId_Max, pulseDuration * 1000.0f, 1, pulseIntensity > 0.5f ? WVR_Intensity_Normal : WVR_Intensity_Strong);
```

Comment:
Since it has five intensities wouldn't it be better to map the 0.0 to 1.0 rang into fifths? That way you get all the intensities. something like:
```
int intensity = ceil(pulseIntensity * 5);
intensity <= 5 ? intensity : 5;
```